packages/client/src/runtime/getPrismaClient.ts

```diff
@@ -1077,12 +1075,12 @@ new PrismaClient({
           // if this `next` was called and there's some more middlewares
           const nextMiddleware = this._middlewares.query.get(++index)
 
-          if (nextMiddleware) {
-            // we pass the modfied params down to the next one, & repeat
-            return nextMiddleware(changedParams, consumer)
-          }
+          // we pass the modified params down to the next one, & repeat
+          if (nextMiddleware) return nextMiddleware(changedParams, consumer)
+          // calling `next` calls the consumer again with the new params
 
-          const changedInternalParams = { ...internalParams, ...params }
+          // before we send the execution request, we use the changed params
+          const changedInternalParams = { ...internalParams, ...changedParams }
```

Comment:
intentional change?
no change in functionality for users?

Reply:
Intentional, yes. It's a bug. Probably went undetected for a while because users mutate `params` in their middleware. But in fact - while that is possible - you should be able to pass a brand new object in the `next(...)` call as well. With the previous logic, it would not be possible.